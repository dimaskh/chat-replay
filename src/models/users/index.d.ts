// Props
export declare const PROP_ID: 'id'
export declare const PROP_USER_NAME: 'user_name'
export declare const PROP_USERNAME: 'username'
export declare const PROP_DISPLAY_NAME: 'display_name'

// Model
export declare const MODEL: 'User'

// Entity
export type Message = {
  [PROP_ID]: number
  [PROP_USER_NAME]?: string
  [PROP_USERNAME]?: string
  [PROP_DISPLAY_NAME]: string
}

// Getters
export declare const getId: GetProperty<typeof PROP_ID>
export declare const getUserName: GetProperty<typeof PROP_USER_NAME>
export declare const getDisplayName: GetProperty<typeof PROP_DISPLAY_NAME>
