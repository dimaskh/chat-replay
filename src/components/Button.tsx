// Libs
import React from 'react'

// Styles
import styled from 'styled-components'

type Props = {
  /** Button title string */
  title: string
  /** Function that handles onClick */
  onClick: () => void
}

export const Button: React.FC<Props> = ({ title = '', onClick }: Props) => (
  <StyledButton onClick={onClick}>{title}</StyledButton>
)

const StyledButton = styled.div`
  color: red;
  cursor: pointer;
  user-select: none;
  background-color: #f18844;
  color: black;
  width: 180px;
`

export default Button
