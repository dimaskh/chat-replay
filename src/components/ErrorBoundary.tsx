// Libs
import React from 'react'

// Styles
import styled from 'styled-components'

export default class ErrorBoundary extends React.PureComponent {
  public state = {
    error: null,
    info: null,
  }

  public componentDidCatch(error: any, info: any) {
    this.setState({ error, info })
  }

  public render() {
    const { error } = this.state

    if (error) {
      return <ErrorBoundaryFallbackComponent />
    }

    return this.props.children
  }
}

const Container = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`

const Message = styled.div`
  padding: 40px;
  border: 2px #78909c solid;
  border-radius: 5px;
  font-size: 24px;
  color: #78909c;
`

const ErrorBoundaryFallbackComponent = () => (
  <Container>
    <Message>
      Some Error Occurred
      <span role="img" aria-label="face-emoji">
        ðŸ˜ž
      </span>
    </Message>
  </Container>
)
