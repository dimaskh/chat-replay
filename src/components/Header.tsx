// Libs
import React from 'react'

// Utils
import styled from 'styled-components'

export type Props = {
  children: string
  medium?: boolean
  small?: boolean
}

export const Header: React.FC<Props> = ({ children, ...props }: Props) => (
  <StyledHeader {...props}>{children}</StyledHeader>
)

export const StyledHeader = styled.header`
  background-color: #282c34;
  height: ${(props: Props) => {
    if (props.medium) return '60px'
    if (props.small) return '40px'
    return '100px'
  }};
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: ${(props: Props) => {
    if (props.medium) return '20px'
    if (props.small) return '12px'
    return '32px'
  }};
  color: white;
  text-align: center;
  border-bottom: 1px solid #737373;
  width: 100%;
`

export default Header
