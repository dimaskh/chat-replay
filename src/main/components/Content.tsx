// Libs
import React from 'react'

// Components
import Chat from 'modules/chat/components/Chat'
import Player from 'modules/player/containers/Player'
import UsersList from 'modules/users/components/UsersList'

// Styles
import styled from 'styled-components'

export const Content: React.FC = () => (
  <StyledContent>
    <Player />
    <UsersList />
    <Chat />
  </StyledContent>
)

const StyledContent = styled.div`
  display: flex;
  flex-direction: row;
  height: 100%;
  background-color: #282c34;
`

export default Content
