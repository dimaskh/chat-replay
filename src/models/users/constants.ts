// Props
export const PROP_ID = 'id'
export const PROP_USER_NAME = 'user_name'
export const PROP_USERNAME = 'username'
export const PROP_DISPLAY_NAME = 'display_name'

// Model
export const MODEL = 'User'
