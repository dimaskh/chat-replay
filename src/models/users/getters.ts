// Composer
import composeGetter from 'utils/composeGetter'

// Constants
import * as props from './constants'

export const getId: Function = composeGetter(props.PROP_ID)
export const getUserName: Function = composeGetter(props.PROP_USER_NAME)
export const getDisplayName: Function = composeGetter(props.PROP_DISPLAY_NAME)
