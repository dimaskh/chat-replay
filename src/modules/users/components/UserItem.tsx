// Libs
import React from 'react'

// Components
import { ChatItem } from 'react-chat-elements'

// Getters
import { getDisplayName } from 'models/users/getters'

// Utils
import { Md5 } from 'ts-md5'

// Types
import { User } from 'models/users'

export type Props = {
  user: User
  date?: string
}

export default class UserItem extends React.PureComponent<Props> {
  public render() {
    const { date, user } = this.props

    return <ChatItem avatar={this.getAvatar()} date={date} title={getDisplayName(user)} subtitle={'Online'} />
  }

  private getAvatar = (): string => {
    const { user } = this.props
    const hash = Md5.hashStr(getDisplayName(user))

    return `https://www.gravatar.com/avatar/${hash}?s=32&d=identicon&r=PG`
  }
}
