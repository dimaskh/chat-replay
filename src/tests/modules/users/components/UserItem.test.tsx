// Libs
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

// Tested component
import UserItem from 'modules/users/components/UserItem'

// Mocks
import { user } from 'tests/mocks/users'

// Types
import { Props } from 'modules/users/components/UserItem'
import { ShallowWrapper } from 'enzyme'

describe('users/UserItem component', () => {
  let component: ShallowWrapper<Props, {}, UserItem>
  let props: Props

  beforeEach(() => {
    props = {
      user,
      date: '2020-04-19T00:00:01.30Z',
    }
    component = shallow<UserItem>(<UserItem {...props} />)
  })

  it('should render correctly', () => {
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should return correct avatar url', () => {
    const instance = component.instance() as any
    expect(instance.getAvatar()).toMatch(/https:\/\/www.gravatar.com\/avatar\//)
  })
})
