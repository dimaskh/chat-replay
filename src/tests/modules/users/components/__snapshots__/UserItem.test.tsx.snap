// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users/UserItem component should render correctly 1`] = `
<t
  alt=""
  avatar="https://www.gravatar.com/avatar/cd00f46ea75c8593fd2a58ae9ed62424?s=32&d=identicon&r=PG"
  avatarFlexible={false}
  date="2020-04-19T00:00:01.30Z"
  dateString={null}
  id=""
  onAvatarError={[Function]}
  onClick={null}
  statusColor={null}
  statusColorType="badge"
  statusText={null}
  subtitle="Online"
  title="Taco Spolsky"
  unread={0}
/>
`;
